code/go/0chain.net/smartcontract/storagesc/readpool.go

```diff
@@ -76,22 +74,13 @@ func (rp *readPool) Decode(p []byte) error {
 	return json.Unmarshal(p, rp)
 }
 
-func (rp *readPool) add(isOwner bool, balance currency.Coin) {
-	if isOwner {
-		rp.OwnerBalance += balance
-		return
-	}
-	rp.VisitorBalance += balance
+func (rp *readPool) add(balance currency.Coin) {
+	rp.Balance += balance
```

Comment:
Call currency.AddCoin() to add coins and check error

Reply:
done